wolfssl/wolfcrypt/port/caam/caam_driver.h

```diff
@@ -138,50 +190,200 @@
+#define CAAM_C1MR        0x8004
 
 
 /* output FIFO  is 16 entries deep and each entry has a two 4 byte registers */
-#define CAAM_FIFOO_MS    CAAM_BASE + 0x87F0
-#define CAAM_FIFOO_LS    CAAM_BASE + 0x87F4
+#define CAAM_FIFOO_MS    0x87F0
+#define CAAM_FIFOO_LS    0x87F4
 
 /* input FIFO is 16 entries deep with each entry having two 4 byte registers
    All data written to it from IP bus should be in big endian format */
-#define CAAM_FIFOI_LS    CAAM_BASE + 0x87E0
+#define CAAM_FIFOI_LS    0x87E0
 
 /* offset of 4 with range 0 .. 13 */
-#define CAAM_CTX1        CAAM_BASE + 0x8100
+#define CAAM_CTX1        0x8100
 #define CAAM_CTRIV       CAAM_CTX1 + 8 /* AES-CTR iv is in 2 and 3 */
 #define CAAM_CBCIV       CAAM_CTX1     /* AES-CBC iv is in 1 and 2 */
 
 
-/* instantiate RNG and create JDKEK, TDKEK, and TDSK key */
-static unsigned int wc_rng_start[] = {
-    CAAM_HEAD | 0x00000006,
-    CAAM_OP | CAAM_CLASS1 | CAAM_RNG | 0x00000004, /* Instantiate RNG handle 0 with TRNG */
-    CAAM_JUMP | 0x02000001,  /* wait for Class1 RNG and jump to next cmd */
-    CAAM_LOAD | 0x00880004,  /* Load to clear written register */
-    0x00000001, /* reset done interrupt */
-    CAAM_OP | CAAM_CLASS1 | CAAM_RNG | 0x00001000   /* Generate secure keys */
+    /* debugging registers */
+#define CAAM_DECORR 0x009C /* used to set DECO into debug mode */
+#define CAAM_DODJR  0x8E00 /* for hung operations */
+#define CAAM_DOJQCR_MS 0x8800
+#define CAAM_DOOPSTA_MS 0x8810 /* DECO operation status register */
+#define CAAM_DODAR  0x8808 /* address of current descriptor */
+#define CAAM_DODESB 0x8A00 /* 64 registers that hold the current descriptor buffer */
+
+
+#define JRINTR_JR0 0x104C
+#define JRINTR_JR1 0x204C
+#define JRINTR_JR2 0x304C
+
+#define CAAM_SINGLE_STEP_MODE 0x40000000
+#define CAAM_STEP 0x80000000
+
+/* Port layer for CAAM driver, functions defined in caam_<env>.c */
+unsigned int CAAM_READ(unsigned int reg);
+void CAAM_WRITE(unsigned int reg, unsigned int in);
+int CAAM_SET_BASEADDR(void);
+unsigned int CAAM_ADR_TO_PHYSICAL(void* in, int inSz);
+void* CAAM_ADR_MAP(unsigned int in, int inSz, unsigned char copy);
+void CAAM_ADR_UNMAP(void* vaddr, unsigned int out, int outSz,
+        unsigned char copy);
+int CAAM_ADR_SYNC(void* vaddr, int sz);
+CAAM_ADDRESS CAAM_ADR_TO_VIRTUAL(CAAM_ADDRESS in, int length);
+
+
+
+    #ifndef WOLFSSL_CAAM_BUFFER
+    #define WOLFSSL_CAAM_BUFFER
+    typedef struct CAAM_BUFFER {
+        int BufferType;
+        CAAM_ADDRESS TheAddress;
+        int Length;
+    } CAAM_BUFFER;
+    #endif
+    unsigned int caamReadRegister(unsigned int reg);
+    void caamWriteRegister(unsigned int reg, unsigned int in);
+    int SynchronousSendRequest(int type, unsigned int args[4], CAAM_BUFFER *buf, int sz);
+    int caamJobRingFree(void);
+
+
+/* Driver API that can be called by caam_<env>.c port layers */
+typedef struct DESCSTRUCT DESCSTRUCT;
+
+int caamKeyCover(DESCSTRUCT *desc, int sz, unsigned int args[4]);
```

Comment:
Should these have `WOLFSSL_LOCAL`?

Reply:
The way the file naming is, all caam_*.h and caam_*.c files are part of the driver that gets compiled apart from wolfSSL. The files with wolfcaam_*.h and wolfcaam_*.c are compiled into the wolfSSL library. Currently this file should not know about WOLFSSL_LOCAL but I can add in a version of local to the driver files.